File: ungs-docsys-bff/src/modules/job-applications/dtos/job-application-response.dto.ts
import { AppUserExtendedResponseDto } from "../../../modules/login/app-user/dtos/app-user-extended-response.dto";
import { JobApplicationPeriodResponseDto } from "../job-application-periods/dtos/job-application-period-response.dto";
import { JobApplicationStatusResponseDto } from "../job-application-statuses/dtos/job-application-status-response.dto";
import { RequirementResponseDto } from "../requirements/dtos/requirement-response.dto";
import { JobProfileLevelResponseDto } from "../job-profile-levels/dtos/job-profile-level.dto";

export class JobApplicationResponseDto {
  id: number;
  code: string;
  title: string;
  description: string;
  jobApplicationStatus: JobApplicationStatusResponseDto;
  jobApplicationPeriod: JobApplicationPeriodResponseDto;
  minApprovers: number;
  reason: string;
  yearPeriod: number;
  active: boolean;
  appUser: AppUserExtendedResponseDto;
  createdDate: Date;
  updatedDate: Date;
  requirements: RequirementResponseDto[];
  jobProfileLevel: JobProfileLevelResponseDto;
}

File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/JobApplicationResponseDto.java
package com.ungs.docsys.dtos;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class JobApplicationResponseDto {
    private Long id;
    private String code;
    private String title;
    private String description;
    private JobApplicationStatusResponseDto jobApplicationStatus;
    private JobApplicationPeriodResponseDto jobApplicationPeriod;
    private Long minApprovers;
    private String reason;
    private Long yearPeriod;
    private Boolean active;
    private AppUserResponseDto appUser;
    private LocalDateTime createdDate;
    private LocalDateTime updatedDate;
    private List<RequirementResponseDto> requirements;
    private JobProfileLevelResponseDto jobProfileLevel;
}
